### PR TITLE
feature(gcp): add support for `gcp_project`

### PR DESCRIPTION
**Pull request by fruch**
we can now run with a diffrent project on demend,
as long as we have the credentials file for it available

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by vponomaryov — APPROVED** [review]

**Review by soyacz — COMMENTED** [review]
Can you describe some background and/or elaborate why do we need this change?

File: sdcm/keystore.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def get_logging_gcp_credentials(self):
         return self.get_json("skilled-adapter-452-4b259a7c0bae.json")
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/keystore.py:57]: yeah, we should throw that code away.

I've fixed the current project file, and we don't need this duplicate of a key no more.

I'll fix this in Sunday

**Comment by fruch** [comment]
> Can you describe some background and/or elaborate why do we need this change?

As part of the issues with are having with GCP, it was asked that we'll be able to run SCT in a fresh new seperate project. This PR enables that.

File: sdcm/keystore.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/keystore.py]: handled in https://github.com/scylladb/scylla-cluster-tests/pull/5011/commits/b19ac74cc30c8229deb3ffd301c84da2596626c6

**Review by roydahan — APPROVED** [review]

**Comment by roydahan** [comment]
Backported to 2022.1 together with several other PRs that introduced dependencies.

**Pull request merged**